deployment/ccip/changeset/cs_deploy_chain.go

```diff
@@ -52,27 +52,108 @@ func DeployChainContractsChangeset(env deployment.Environment, c DeployChainCont
 }
 
 type DeployChainContractsConfig struct {
-	ChainSelectors    []uint64
-	HomeChainSelector uint64
+	HomeChainSelector      uint64
+	ContractParamsPerChain map[uint64]ContractParams
 }
 
 func (c DeployChainContractsConfig) Validate() error {
-	for _, cs := range c.ChainSelectors {
+	if err := deployment.IsValidChainSelector(c.HomeChainSelector); err != nil {
+		return fmt.Errorf("invalid home chain selector: %d - %w", c.HomeChainSelector, err)
+	}
+	for cs, args := range c.ContractParamsPerChain {
 		if err := deployment.IsValidChainSelector(cs); err != nil {
 			return fmt.Errorf("invalid chain selector: %d - %w", cs, err)
 		}
+		if err := args.Validate(); err != nil {
+			return fmt.Errorf("invalid contract args for chain %d: %w", cs, err)
+		}
 	}
-	if err := deployment.IsValidChainSelector(c.HomeChainSelector); err != nil {
-		return fmt.Errorf("invalid home chain selector: %d - %w", c.HomeChainSelector, err)
+	return nil
+}
+
+type ContractParams struct {
```

Comment:
To explicitly mention that it's Chain contracts related
```suggestion
type ChainContractParams struct {
```